routes/login/login.controller.js

```diff
@@ -57,6 +57,18 @@ const postSIN = (req, res) => {
   let sin = req.body.sin || null
   req.session.sin = sin
 
+  const errors = validationResult(req)
+
+  if (!errors.isEmpty()) {
+    // clear session
+    req.session = null
+
+    return res.status(422).render('login/sin', {
+      data: { code: req.body.code } || {},
```

Comment:
To preserve the value that's passed in by the user (instead of clearing it), let's change this to `sin: req.body.sin` instead.